main.go

```diff
@@ -36,46 +36,68 @@ type root struct {
 // StevEx: What are your thoughts on using "this" as the reciever. I've seen some differing opinions on the internet
 // also is the reciever a pointer to root? is that what r *root means? That r "is the type pointer to root" and when we call the method we are "passing"
 // the currently instantiated struct?  My main point of confusion is that in newTransaction() we seem to be passing an argument but most other methods seem to pass
-// ---------------------------------------------------------
-// commands should be methods others can be functions
-
-func repl() {
+func repl(root root) {
 
 	reader := bufio.NewReader(os.Stdin)
 	input, _ := reader.ReadString('\n')
-	parseCommand(input)
+	root.parseCommand(input)
 }
 
 func (r *root) read(key string) (string, error) {
-	if r.db[key] != 0 int {
-		return "", r.db[key]
-	}
 	displayValue := ""
+	if r.transactions == nil {
+		if value, ok := r.db[key]; ok {
+			return value, nil
+		}
+
+		if r.transactions == nil {
```

Comment:
Hmm, looks like you're checking `if r.transactions == nil` on line 48, and then again here on line 53. So it seems like it should always be `nil` here? What do you think about doing something like:

```
	if r.transactions == nil {
		if value, ok := r.db[key]; ok {
			return value, nil
		}

		return displayValue, errors.New("That key was not found")
	}
```

Reply:
Right, my second if statement is unreachable as currently written. I'm not sure why I started a nested if statement and then just wrote a third one anyway!